### PR TITLE
Finish filter functions and Topic question form functionality

### PR DESCRIPTION
**Pull request by ajaguasin**

### CODE REVIEW

File: js/components/AskButton/AskButton.js [hunk 1]
```diff
@@ -19,7 +19,9 @@ const AskButton = props => {
       <TouchableOpacity
         style={styles.questionButton}
         onPress={() => {
-          props.navigation.navigate("CommunityQuestion");
+          props.navigation.navigate("CommunityQuestion", {
+            categoryID: props.categoryID
+          });
         }}
       >
         <Text style={styles.questionButtonText}>Ask a question</Text>
```
Comment (siddharthparmar7) [inline js/components/AskButton/AskButton.js:27]: Please add the prop-types.

File: js/components/CommunityQuesitonForm/CommunityQuestionForm.js [hunk 2]
```diff
@@ -1,76 +1,127 @@
...
+      body: JSON.stringify(body)
+    });
+    console.log(res);
+  };
+
+  handleForm = values => {
+    const body = values;
+    body.category_id = this.state.category_id;
+    body.is_anonymous = this.state.isAnon;
+    console.log(body);
```
Comment (siddharthparmar7) [inline js/components/CommunityQuesitonForm/CommunityQuestionForm.js]: remove console logs.

File: js/components/CommunityQuesitonForm/CommunityQuestionForm.js [hunk 3]
```diff
@@ -1,76 +1,127 @@
...
+    const token =
+      "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczpcL1wvYXBpLnNwZWFrYm94LXN0YWdpbmcuY2FcL2FwaVwvYXV0aFwvbG9naW4iLCJpYXQiOjE1NDQ2NTQyMzgsImV4cCI6MTU0NDY1NzgzOCwibmJmIjoxNTQ0NjU0MjM4LCJqdGkiOiJ6dzFFWkEzcUpxSkZVNktxIiwic3ViIjoxLCJwcnYiOiI4N2UwYWYxZWY5ZmQxNTgxMmZkZWM5NzE1M2ExNGUwYjA0NzU0NmFhIn0.bw7f_UwpbfdFwloC7_EcJ1xB6CJ4GUFb4X718l4k2aE";
+    const res = await apiPromise("/topics", {
+      method: "POST",
+      headers: {
+        Authorization: `Bearer${[token]}`
+      },
+      body: JSON.stringify(body)
+    });
+    console.log(res);
```
Comment (siddharthparmar7) [inline js/components/CommunityQuesitonForm/CommunityQuestionForm.js]: remove console logs.

File: js/components/CommunityQuesitonForm/CommunityQuestionForm.js [hunk 4]
```diff
@@ -1,76 +1,127 @@
...
+    super(props);
+    this.state = {
+      category_id: this.props.navigation.getParam("categoryID"),
+      isAnon: false
+    };
+  }
+
+  createTopic = async body => {
+    const token =
+      "eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJodHRwczpcL1wvYXBpLnNwZWFrYm94LXN0YWdpbmcuY2FcL2FwaVwvYXV0aFwvbG9naW4iLCJpYXQiOjE1NDQ2NTQyMzgsImV4cCI6MTU0NDY1NzgzOCwibmJmIjoxNTQ0NjU0MjM4LCJqdGkiOiJ6dzFFWkEzcUpxSkZVNktxIiwic3ViIjoxLCJwcnYiOiI4N2UwYWYxZWY5ZmQxNTgxMmZkZWM5NzE1M2ExNGUwYjA0NzU0NmFhIn0.bw7f_UwpbfdFwloC7_EcJ1xB6CJ4GUFb4X718l4k2aE";
```
Comment (siddharthparmar7) [inline js/components/CommunityQuesitonForm/CommunityQuestionForm.js:25]: how far is your team from storing this into realm?

File: js/components/CommunityQuesitonForm/CommunityQuestionForm.js [hunk 5]
```diff
@@ -1,76 +1,127 @@
...
+  };
+
+  render() {
+    return (
+      <Form
+        style={styles.form}
+        onSubmit={values => {
+          this.handleForm(values);
+          // const getPosts = this.props.navigation.getParam("getPosts");
+          // getPosts();
```
Comment (siddharthparmar7) [inline js/components/CommunityQuesitonForm/CommunityQuestionForm.js]: please remove these comments.

File: js/components/CommunityQuesitonForm/CommunityQuestionForm.js [hunk 6]
```diff
@@ -1,76 +1,127 @@
...
+                <Picker.Item label="Mental Health" value={2} />
+                <Picker.Item label="Work" value={3} />
+                <Picker.Item label="School" value={4} />
+                <Picker.Item label="Relationships" value={5} />
+              </Picker>
+            </React.Fragment>
+
+            <View style={styles.switchContainer}>
+              <Switch
+                // {...input}
```
Comment (siddharthparmar7) [inline js/components/CommunityQuesitonForm/CommunityQuestionForm.js]: please remove this comment.

File: js/components/CommunityQuesitonForm/styles.js [hunk 7]
```diff
@@ -12,34 +13,51 @@ const styles = StyleSheet.create({
...
     borderRadius: 20,
-    width: Dimensions.get("window").width / 3
+    width: width / 3
   },
 
   picker: {
-    flex: 1,
-    justifyContent: "center"
+    height: height / 6,
+
```
Comment (siddharthparmar7) [inline js/components/CommunityQuesitonForm/styles.js:41]: weird gap? please use prettier.

File: js/components/FilterButton/FilterButton.js [hunk 8]
```diff
@@ -16,7 +16,11 @@ const FilterButton = props => {
       ]}
       style={styles.filterButton}
     >
-      <TouchableOpacity>
+      <TouchableOpacity
+        onPress={() => {
+          props.getPosts(props.filterName);
+        }}
+      >
```
Comment (siddharthparmar7) [inline js/components/FilterButton/FilterButton.js:23]: please add your prop type validation.

File: js/screens/CommunityCategory/CommunityCategory.js [hunk 9]
```diff
@@ -1,5 +1,11 @@
...
+import {
+  View,
+  Text,
+  FlatList,
+  TouchableOpacity,
+  ImageBackground
+} from "react-native";
 import PropTypes from "prop-types";
 import LinearGradient from "react-native-linear-gradient";
 import styles from "./styles";
```
Comment (siddharthparmar7) [inline js/screens/CommunityCategory/CommunityCategory.js:10]: please remove unnecessary imports and add prop type validation.

File: js/screens/CommunityCategory/CommunityCategoryContainer.js [hunk 10]
```diff
@@ -1,9 +1,10 @@
 import React, { Component } from "react";
 import CommunityCategory from "./CommunityCategory";
 import { apiPromise } from "../../lib/API/";
+import { withNavigationFocus } from "react-navigation";
 import { Text } from "react-native";
```
Comment (siddharthparmar7) [inline js/screens/CommunityCategory/CommunityCategoryContainer.js]: please remove unnecessary imports

File: js/screens/CommunityQuestion/CommunityQuestion.js [hunk 11]
```diff
@@ -1,6 +1,6 @@
 import React from "react";
 import PropTypes from "prop-types";
-import { View, Text, TouchableOpacity } from "react-native";
+import { ScrollView, View, Text, TouchableOpacity } from "react-native";
 import CommunityQuestionForm from "../../components/CommunityQuesitonForm";
 import LinearGradient from "react-native-linear-gradient";
 import styles from "./styles";
```
Comment (siddharthparmar7) [inline js/screens/CommunityQuestion/CommunityQuestion.js:6]: add prop type validations

File: js/screens/CommunityQuestion/CommunityQuestionContainer.js [hunk 12]
```diff
@@ -1,7 +1,7 @@
 import React, { Component } from "react";
 import CommunityQuestion from "./CommunityQuestion";
 
-export default class CommunityQuestionContainer extends Component {
+class CommunityQuestionContainer extends Component {
   constructor(props) {
     super(props);
   }
```
Comment (siddharthparmar7) [inline js/screens/CommunityQuestion/CommunityQuestionContainer.js:8]: remove the constructor if you don't need it.

File: js/screens/CommunityQuestion/CommunityQuestionContainer.js [hunk 13]
```diff
@@ -10,6 +10,8 @@ export default class CommunityQuestionContainer extends Component {
   };
 
   render() {
-    return <CommunityQuestion />;
+    return <CommunityQuestion navigation={this.props.navigation} />;
```
Comment (siddharthparmar7) [inline js/screens/CommunityQuestion/CommunityQuestionContainer.js:14]: add prop type validation.

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Please make the requested changes. Thank you.

File: js/components/CommunityQuesitonForm/CommunityQuestionForm.js (hunk 4 above)
Comment (ajaguasin) [reply js/components/CommunityQuesitonForm/CommunityQuestionForm.js:25]: Devin is currently working on Realm and Context, should be done by friday latest

File: js/screens/CommunityQuestion/CommunityQuestionContainer.js (hunk 12 above)
Comment (ajaguasin) [reply js/screens/CommunityQuestion/CommunityQuestionContainer.js:8]: it has a proptype of navigation, I added it and kept the constructor and props

File: js/components/AskButton/AskButton.js [hunk 14]
```diff
@@ -28,6 +30,9 @@ const AskButton = props => {
   );
 };
 
-AskButton.propTypes = {};
+AskButton.propTypes = {
+  navigation: PropTypes.object.isRequired,
+  categoryID: PropTypes.number
```
Comment (siddharthparmar7) [inline js/components/AskButton/AskButton.js:35]: If it is not required, you should have a default prop for that.

File: js/components/CommunityQuesitonForm/CommunityQuestionForm.js (hunk 4 above)
Comment (siddharthparmar7) [reply js/components/CommunityQuesitonForm/CommunityQuestionForm.js:25]: ok great!

**Review by siddharthparmar7 — CHANGES_REQUESTED** [review]
Seems like some of the changes I asked for are not done. Please make the changes and let me know. Thank you.

File: js/components/CommunityQuesitonForm/CommunityQuestionForm.js [hunk 15]
```diff
@@ -1,76 +1,123 @@
...
-    />
-  );
-};
+class CommunityQuestionForm extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      category_id: this.props.navigation.getParam("categoryID"),
+      isAnon: false
+    };
```
Comment (siddharthparmar7) [inline js/components/CommunityQuesitonForm/CommunityQuestionForm.js:15]: we don't assign props to a state in the constructor. Please update your state in the component lifecycle method.

File: js/components/CommunityQuesitonForm/CommunityQuestionForm.js (hunk 6 above)
Comment (siddharthparmar7) [reply js/components/CommunityQuesitonForm/CommunityQuestionForm.js]: it's still here. Please remove it.

**Review by siddharthparmar7 — APPROVED** [review]
Nice work

**Pull request merged**